src/ruby.ts

```diff
@@ -120,7 +121,8 @@ function registerLinters(ctx: ExtensionContext) {
 	}
 
 	ctx.subscriptions.push(vscode.window.onDidChangeActiveTextEditor(executeLinting));
-	ctx.subscriptions.push(vscode.workspace.onDidChangeTextDocument(executeLinting));
+	// Debounce linting to prevent running on every keypress, only run when typing has stopped
+	ctx.subscriptions.push(vscode.workspace.onDidChangeTextDocument(debounce(executeLinting, 500)));
```

Comment:
You might make the debounce delay configurable?  Some of the other delays (eg autocomplete recommendations) are configurable.

Reply:
I hardcoded it as I didn't think there would be too much need to change it, and didn't want to needlessly add extra config options.

If it would be useful for some to be able to configure it though, I'm happy to do so.

Reply:
Ultimately up to maintainers, but any time I see a delay I like it to be configurable (unit testing, different user preferences, etc).